api/extra.go

```diff
@@ -49,14 +55,102 @@ func ResolveDidToHandle(ctx context.Context, xrpcc *xrpc.Client, res did.Resolve
 		return "", "", fmt.Errorf("our XRPC client is authed for a different pds (%s != %s)", svc.ServiceEndpoint, xrpcc.Host)
 	}
 
-	verdid, err := comatproto.IdentityResolveHandle(ctx, xrpcc, handle)
+	verdid, err := hr.ResolveHandleToDid(ctx, handle)
 	if err != nil {
 		return "", "", err
 	}
 
-	if verdid.Did != udid {
+	if verdid != udid {
 		return "", "", fmt.Errorf("pds server reported different did for claimed handle")
 	}
 
 	return handle, svc.ServiceEndpoint, nil
 }
+
+type HandleResolver interface {
+	ResolveHandleToDid(ctx context.Context, handle string) (string, error)
+}
+
+type ProdHandleResolver struct {
+}
+
+func (dr *ProdHandleResolver) ResolveHandleToDid(ctx context.Context, handle string) (string, error) {
+	c := http.DefaultClient
```

Comment:
This HTTP client in particular should have a short timeout, like 5-10 seconds, I think. Not formalized anywhere, but don't want to spend a bunch of time waiting for what should be a very quick response.